src/Tools/Source/CompilerGeneratorTools/Source/BoundTreeGenerator/BoundNodeClassWriter.cs

```diff
@@ -471,8 +471,23 @@ private void WriteConstructorWithHasErrors(TreeType node, bool isPublic, bool ha
                                 WriteLine("this.{0} = {1};", field.Name, FieldNullHandling(node, field.Name) == NullHandling.Always ? "null" : ToCamelCase(field.Name));
                             }
                         }
+
+                        bool hasValidate = HasValidate(node);
+
+                        if (hasValidate)
+                        {
+                            WriteLine("Validate();");
+                        }
+
                         Unbrace();
                         Blank();
+
+                        if (hasValidate)
+                        {
+                            WriteLine("private partial void Validate();");
```

Comment:
💡  Would be nice to add something like `[Conditional("DEBUG")] partial void PostInitValidation();` for *all* nodes

Reply:
I agree, it feels like the design of partial methods is such that we shouldn't need to introduce new flags to BoundNodes.xml or branches in this logic. Just unconditionally emit the declaration part and the call, then trust the compiler to eliminate the methods and calls to methods that lack implementations.